pkgs/development/python-modules/textual-dev/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, aiohttp
+, buildPythonPackage
+, click
+, fetchFromGitHub
+, msgpack
+, poetry-core
+, pythonOlder
+, textual
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "textual-dev";
+  version = "1.2.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "Textualize";
+    repo = "textual-dev";
+    rev = "fb5aa869784895066dbd0ec850d92ac9205cde73";
+    hash = "sha256-WO2uYw9rUvEMlhir3T8UA8VXGN/BU7Arn4Nxim1j5k0=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    aiohttp
+    click
+    msgpack
+    textual
+    typing-extensions
+  ];
+
```

Comment:
Could you please enable tests with pytestCheckHook?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

Reply:
Done I believe, and all the tests passed :)